react-native/react/native/remote-manager.desktop.js

```diff
@@ -129,28 +143,67 @@ class RemoteManager extends Component {
     })
   }
 
+  showUpdatePromptComponents () {
+    const {showUpdateState} = this.props
+
+    if (showUpdateState.closed) {
+      return
+    }
+
+    const windowOpts = {
+      width: 480, height: 685 + 20 /* TEMP workaround for header mouse clicks in osx */,
+      resizable: true,
+      fullscreen: false,
+      show: false,
+      frame: false
+    }
+
+    return (
+      <RemoteComponent
+        windowsOpts={windowOpts}
+        waitForState
+        component='update'
+        substore='update'
+        onCancel={() => this.props.updateOnCancel()}
+        onSkip={() => this.props.updateOnSkip()}
+        onSnooze={() => this.props.updateOnSnooze()}
+        onUpdate={() => this.props.updateOnUpdate()}
+        onRemoteClose={() => this.props.updateOnCancel()}
+        setAlwaysUpdate={alwaysUpdate => this.props.setAlwaysUpdate(alwaysUpdate)}
+      />
+    )
+  }
+
   render () {
     return (
       <div>
         {Object.keys(this.state.popups).filter(username => !this.props.trackers[username].closed).map(username => this.state.popups[username])}
         {this.pinentryRemoteComponents()}
+        {this.showUpdatePromptComponents()}
       </div>
     )
   }
 }
 
 RemoteManager.propTypes = {
-  pinentryOnCancel: React.PropTypes.any,
-  pinentryOnSubmit: React.PropTypes.any,
-  registerPinentryListener: React.PropTypes.any,
-  registerIdentifyUi: React.PropTypes.any,
+  pinentryOnCancel: React.PropTypes.func,
```

Comment:
this is all kinda painful. can't we just bring in the actions into the components themselves? @MarcoPolo